src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -167,24 +168,43 @@ C_FUNC(\Name\()_End):
 
 .macro EPILOG_WITH_TRANSITION_BLOCK_RETURN
 
-        free_stack __PWTB_StackAlloc
-        POP_CALLEE_SAVED_REGISTERS
-        free_stack 16
-        bx lr
+	// Do not restore r7, lr until EPILOG.
+
+	add r6, sp, #(__PWTB_StackAlloc)
+	ldm r6, {r4-r5}
+
+	add r6, sp, #((__PWTB_StackAlloc) + 4 * 4)
+	ldm r6, {r8-r11}
+
+	ldr r6, [sp, #((__PWTB_StackAlloc) + 8)]
+
+	free_stack ((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+	EPILOG_POP "{r7, pc}"
 
 .endm
 
 .macro EPILOG_WITH_TRANSITION_BLOCK_TAILCALL
 
         .if (__PWTB_SaveFPArgs == 1)
-                add r6, sp, #(__PWTB_FloatArgumentRegisters)
+                mov r6, sp
                 vldm r6, {d0-d7}
         .endif
 
-        free_stack __PWTB_StackAlloc
-        POP_CALLEE_SAVED_REGISTERS
-        POP_ARGUMENT_REGISTERS
-        
+	// Do not restore r7, lr until EPILOG.
+
+	add r6, sp, #((__PWTB_StackAlloc) + 9 * 4)
+	ldm r6, {r0-r3}
+
+	add r6, sp, #(__PWTB_StackAlloc)
+	ldm r6, {r4-r5}
+
+	add r6, sp, #((__PWTB_StackAlloc) + 4 * 4)
+	ldm r6, {r8-r11}
+
+	ldr r6, [sp, #((__PWTB_StackAlloc) + 8)]
+
+	free_stack ((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+	EPILOG_POP "{r7, lr}"
```

Comment:
For me now, after the last EPILOG_POP, gdb is not showing the call stack trace anymore. 

Locally I played a trick by combining the EPILOG_WITH_TRANSITION_BLOCK_TAIL with EPILOG_BRANCH_REG, so store the register (usually r12 in where lr was stored on the stack) and then doing EPILOG_POP {r7, pc}. It worked better but still not satisfactory.